File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyPullCallback.java
package com.kinvey.androidTest.callback;

import com.kinvey.android.sync.KinveyPullCallback;
import com.kinvey.java.model.KinveyPullResponse;

import java.util.concurrent.CountDownLatch;

/**
 * Created by yuliya on 09/14/17.
 */

public class CustomKinveyPullCallback implements KinveyPullCallback {

    private CountDownLatch latch;
    private KinveyPullResponse result;
    private Throwable error;

    public CustomKinveyPullCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(KinveyPullResponse result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    void finish() {
        latch.countDown();
    }

    public KinveyPullResponse getResult() {
        return result;
    }

    public Throwable getError() {
        return error;
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/callback/CustomKinveyPullCallback.kt
package com.kinvey.androidTest.callback

import com.kinvey.android.sync.KinveyPullCallback
import com.kinvey.java.model.KinveyPullResponse
import java.util.concurrent.CountDownLatch

/**
 * Created by yuliya on 09/14/17.
 */

class CustomKinveyPullCallback(private val latch: CountDownLatch) : KinveyPullCallback {
    var result: KinveyPullResponse? = null
        private set
    var error: Throwable? = null
        private set

    override fun onSuccess(result: KinveyPullResponse) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable) {
        this.error = error
        finish()
    }

    internal fun finish() {
        latch.countDown()
    }

}